src/backend/distributed/commands/create_distributed_table.c

```diff
@@ -349,6 +349,15 @@ ConvertToDistributedTable(Oid relationId, char *distributionColumnName,
 								  "foreign tables.")));
 	}
 
+#if (PG_VERSION_NUM >= 100000)
+	if (relation->rd_rel->relispartition)
+	{
+		ereport(ERROR, (errcode(ERRCODE_WRONG_OBJECT_TYPE),
+						errmsg("cannot distribute relation: %s", relationName),
+						errdetail("Distributing partition tables is unsupported.")));
+	}
```

Comment:
It's sufficient because partitioned tables are caught by the above check (see @onderkalaci's original comment about this); they're not regular or foreign tables, so the existing error is straightforward.

If you care, I can add a special error for PostgreSQL 10 which looks for that type, but at the moment we don't handle them specially for 9.5 or 9.6 and the above "is this regular or foreign?" check catches them just fine in 10.